src/fileimpl.h

```diff
@@ -86,6 +89,8 @@ namespace zim
       article_index_t getCountArticles() const { return article_index_t(header.getArticleCount()); }
 
 
+      article_index_type getArticleByClusterOrder(article_index_type idx) const;
```

Comment:
Same remarks here about the api. Other method are `findx*` and return a pair.
It should be the same (`findxByClusterOrder`)

Reply:
Ok. If we are returing a pair that holds the success flag on the first position we can avoid throwing an exception. Then in the caller we can return an empty Article.